drivers/sensor/adt7420/adt7420_trigger.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2018 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <gpio.h>
+#include <i2c.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <sensor.h>
+#include "adt7420.h"
+
+extern struct adt7420_data adt7420_driver;
+
+static void adt7420_thread_cb(void *arg)
+{
+	struct device *dev = arg;
+	struct adt7420_data *drv_data = dev->driver_data;
+	u8_t status;
+
+	/* Clear the status */
+	if (i2c_reg_read_byte(drv_data->i2c, drv_data->i2c_addr,
+			      ADT7420_REG_STATUS, &status) < 0)
+		return;
+
+	if (drv_data->th_handler != NULL)
+		drv_data->th_handler(dev, &drv_data->th_trigger);
+
+	gpio_pin_enable_callback(drv_data->gpio, CONFIG_ADT7420_GPIO_PIN_NUM);
+}
+
+static void adt7420_gpio_callback(struct device *dev,
+				  struct gpio_callback *cb, u32_t pins)
+{
+	struct adt7420_data *drv_data =
+		CONTAINER_OF(cb, struct adt7420_data, gpio_cb);
+
+	gpio_pin_disable_callback(dev, CONFIG_ADT7420_GPIO_PIN_NUM);
+
+#if defined(CONFIG_ADT7420_TRIGGER_OWN_THREAD)
+	k_sem_give(&drv_data->gpio_sem);
+#elif defined(CONFIG_ADT7420_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&drv_data->work);
+#endif
+}
+
+#if defined(CONFIG_ADT7420_TRIGGER_OWN_THREAD)
+static void adt7420_thread(int dev_ptr, int unused)
+{
+	struct device *dev = INT_TO_POINTER(dev_ptr);
+	struct adt7420_data *drv_data = dev->driver_data;
+
+	ARG_UNUSED(unused);
+
+	while (true) {
+		k_sem_take(&drv_data->gpio_sem, K_FOREVER);
+		adt7420_thread_cb(dev);
+	}
+}
+
+#elif defined(CONFIG_ADT7420_TRIGGER_GLOBAL_THREAD)
+static void adt7420_work_cb(struct k_work *work)
+{
+	struct adt7420_data *drv_data =
+		CONTAINER_OF(work, struct adt7420_data, work);
+	adt7420_thread_cb(drv_data->dev);
+}
+#endif
+
+int adt7420_trigger_set(struct device *dev,
+			const struct sensor_trigger *trig,
+			sensor_trigger_handler_t handler)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+
+	gpio_pin_disable_callback(drv_data->gpio, CONFIG_ADT7420_GPIO_PIN_NUM);
+
+	if (trig->type == SENSOR_TRIG_THRESHOLD) {
+		drv_data->th_handler = handler;
+		drv_data->th_trigger = *trig;
+	} else {
+		SYS_LOG_ERR("Unsupported sensor trigger");
+		return -ENOTSUP;
+	}
+
+	gpio_pin_enable_callback(drv_data->gpio, CONFIG_ADT7420_GPIO_PIN_NUM);
+
+	return 0;
+}
+
+int adt7420_init_interrupt(struct device *dev)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+
+	drv_data->gpio = device_get_binding(CONFIG_ADT7420_GPIO_DEV_NAME);
```

Comment:
`CONFIG_ADT7420_GPIO_DEV_NAME` and `CONFIG_ADT7420_GPIO_PIN_NUM` can also be stored in the const config structure

Reply:
will do.